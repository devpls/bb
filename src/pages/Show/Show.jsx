import React, { useEffect } from 'react';
import { BEMHelper } from '@app/utils/bemHelper';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { showsSelector } from '@app/store/selectors';
import { fetchShowViewAction } from '@app/store/actions';
import { Card, Spinner } from '@app/components';
import './Show.scss';

const classes = new BEMHelper('show-page');

export const Show = () => {
  const dispatch = useDispatch();
  const { show, episodes, isLoading, router } = useSelector(showsSelector);
  const id = router?.params?.id;
  useEffect(() => {
    if (!id) {
      return;
    }
    dispatch(fetchShowViewAction({ id }));
  }, [dispatch, id]);

  const renderEpisodeList = (data) => {
    return data?.map((episode) => {
      return (
        <li key={episode.id} className={classes('episode')}>
          <Link to={`/episodes/${episode.id}`} className={classes('episode-link')}>
            {episode.name}
          </Link>
        </li>
      );
    });
  };
  const renderEpisodes = () => {
    return episodes.map((season, index) => {
      const seasonNumber = index + 1;
      return (
        <div className={classes('season')}>
          <h3 className={classes('season-title')}>
            <span>Season:</span>
            {seasonNumber}
          </h3>
          <ul className={classes('season-list')}>{renderEpisodeList(season)}</ul>
        </div>
      );
    });
  };
  // Ideally, the spinner should work at the App level or be part of the page layout. But this will require more logic complexity (e.g. creating an additional action from requestEpic), so please forgive this copy/paste.
  const renderContent = () => {
    if (isLoading) {
      return <Spinner />;
    }
    return (
      <div className={classes('')}>
        <Card image={show?.image?.original} description={show?.summary} title={show?.name} />
        {episodes.length > 0 && (
          <>
            <h2 className={classes('title')}>Episodes:</h2>
            {renderEpisodes()}
          </>
        )}
      </div>
    );
  };

  return <>{renderContent()}</>;
};
