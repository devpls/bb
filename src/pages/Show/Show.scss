.bb-show-page {
  &__title {
    font-size: 20px;
    margin-top: 15px;
  }

  &__list {
    margin-top: 15px;
  }

  &__season {
    counter-reset: list;
    padding: 10px;
  }

  &__episode {
    margin-top: 10px;
    padding-left: 25px;
    position: relative;
    z-index: 0;

    & ~ & {
      margin-top: 5px;
    }

    &::before {
      content: counter(list) '.';
      counter-increment: list;
      left: 0;
      position: absolute;
      text-align: right;
      top: 0;
      width: 20px;
    }
  }

  &__season-title {
    span {
      margin-right: 5px;
    }
  }
}
