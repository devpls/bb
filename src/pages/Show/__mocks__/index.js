export { show } from './show';
export { showEpisodes } from './showEpisodes';
export { router } from './router';
