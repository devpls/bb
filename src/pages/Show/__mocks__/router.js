export const router = {
  location: {
    pathname: '/shows/6771',
    search: '',
    hash: '',
    key: 'nmxul5',
    query: {},
  },
  action: 'POP',
};
