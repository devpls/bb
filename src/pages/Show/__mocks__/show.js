export const show = {
  data: {
    id: 6771,
    url: 'http://www.tvmaze.com/shows/6771/the-powerpuff-girls',
    name: 'The Powerpuff Girls',
    type: 'Animation',
    language: 'English',
    genres: ['Comedy', 'Action', 'Science-Fiction'],
    status: 'Running',
    runtime: 15,
    premiered: '2016-04-04',
    officialSite: 'https://www.cartoonnetwork.com/video/powerpuff-girls/index.html',
    schedule: {
      time: '17:30',
      days: ['Sunday'],
    },
    rating: {
      average: 6.2,
    },
    weight: 43,
    network: {
      id: 11,
      name: 'Cartoon Network',
      country: {
        name: 'United States',
        code: 'US',
        timezone: 'America/New_York',
      },
    },
    webChannel: null,
    externals: {
      tvrage: null,
      thetvdb: 307473,
      imdb: 'tt4718304',
    },
    image: {
      medium: 'http://static.tvmaze.com/uploads/images/medium_portrait/60/151357.jpg',
      original: 'http://static.tvmaze.com/uploads/images/original_untouched/60/151357.jpg',
    },
    summary:
      "<p>The city of Townsville may be a beautiful, bustling metropolis, but don't be fooled! There's evil afoot! And only three things can keep the bad guys at bay: Blossom, Bubbles and Buttercup, three super-powered little girls, known to their fans (and villains everywhere) as <b>The Powerpuff Girls</b>. Juggling school, bedtimes, and beating up giant monsters may be daunting, but together the Powerpuff Girls are up to the task. Battling a who's who of evil, they show what it really means to \"fight like a girl.\"</p>",
    updated: 1573685920,
    _links: {
      self: {
        href: 'http://api.tvmaze.com/shows/6771',
      },
      previousepisode: {
        href: 'http://api.tvmaze.com/episodes/1662010',
      },
    },
  },
  isLoading: false,
  error: null,
};
