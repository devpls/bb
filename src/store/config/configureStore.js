import { createStore, applyMiddleware } from 'redux';
import { createEpicMiddleware } from 'redux-observable';
import { composeWithDevTools } from 'redux-devtools-extension';
import { routerMiddleware } from 'connected-react-router';
import { history } from '@app/constants';
import { createRootReducer } from '../reducers';
import { rootEpic } from '../epics';

const epicMiddleware = createEpicMiddleware();

const configureStore = () => {
  const store = createStore(
    createRootReducer(history),
    composeWithDevTools(applyMiddleware(routerMiddleware(history), epicMiddleware)),
  );

  epicMiddleware.run(rootEpic);

  return store;
};

export default configureStore;
