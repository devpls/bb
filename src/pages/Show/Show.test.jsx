import React from 'react';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Show } from './Show';
import { show, showEpisodes, router } from './__mocks__';

const mockStore = configureStore();
const store = mockStore({
  show,
  showEpisodes,
  router,
});
const history = createMemoryHistory();

describe('Show page', () => {
  it('Should be equal to mock', () => {
    const { container } = render(
      <Provider store={store}>
        <Router history={history}>
          <Show />
        </Router>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
