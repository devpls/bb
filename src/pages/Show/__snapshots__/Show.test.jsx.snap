// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show page Should be equal to mock 1`] = `
<div>
  <div
    class="bb-show-page"
  >
    <div
      class="bb-card"
    >
      <h2
        class="bb-card__title"
      >
        The Powerpuff Girls
      </h2>
      <div
        class="bb-card__content"
      >
        <div
          class="bb-card__image"
        >
          <img
            alt="The Powerpuff Girls"
            src="http://static.tvmaze.com/uploads/images/original_untouched/60/151357.jpg"
          />
        </div>
        <p
          class="bb-card__description"
        >
          <p>
            The city of Townsville may be a beautiful, bustling metropolis, but don't be fooled! There's evil afoot! And only three things can keep the bad guys at bay: Blossom, Bubbles and Buttercup, three super-powered little girls, known to their fans (and villains everywhere) as 
            <b>
              The Powerpuff Girls
            </b>
            . Juggling school, bedtimes, and beating up giant monsters may be daunting, but together the Powerpuff Girls are up to the task. Battling a who's who of evil, they show what it really means to "fight like a girl."
          </p>
        </p>
      </div>
    </div>
    <h2
      class="bb-show-page__title"
    >
      Episodes:
    </h2>
    <div
      class="bb-show-page__season"
    >
      <h3
        class="bb-show-page__season-title"
      >
        <span>
          Season:
        </span>
        1
      </h3>
      <ul
        class="bb-show-page__season-list"
      >
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/657308"
          >
            Escape from Monster Island
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/657309"
          >
            Princess Buttercup
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/691590"
          >
            The Stayover
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/691591"
          >
            Painbow
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/691592"
          >
            Horn, Sweet Horn
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/691593"
          >
            Man Up
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/691594"
          >
            Bye Bye, Bellum
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/691595"
          >
            Little Octi Lost
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/691596"
          >
            Strong Armed
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/691597"
          >
            Power Up Puff
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/691598"
          >
            Tiara Trouble
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/691600"
          >
            The Wrinklegruff Gals
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/691601"
          >
            Arachno Romance
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/691602"
          >
            Puffdora's Box
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/770609"
          >
            Blue Ribbon Blues
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/772889"
          >
            Frenemy
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/777840"
          >
            Once Upon a Townsville
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/805275"
          >
            Man Up 2: Still Man-ing
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/820919"
          >
            Viral Spiral
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/831263"
          >
            Bubbles of the Opera
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/831264"
          >
            Sister Sitter
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/882630"
          >
            Odd Bubbles Out
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/904659"
          >
            Presidential Punchout
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/917876"
          >
            Cheep Thrills
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/904660"
          >
            Fashion Forward
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/904661"
          >
            In The Garden of Good and Eddie
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/904662"
          >
            Road Trippin'
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/904663"
          >
            The Big Sleep
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/904664"
          >
            The Secret Life of Blossom Powerpuff
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/942511"
          >
            Halt and Catch Silico
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/948697"
          >
            Secret Swapper of Doom
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/948698"
          >
            Rainy Day
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/948699"
          >
            The Squashening
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/961257"
          >
            Electric Buttercup
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/979337"
          >
            Professor Proofed
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/979338"
          >
            Poorbucks
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/982635"
          >
            Snow Month
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/982636"
          >
            Somewhere Over the Swingset
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/998421"
          >
            People Pleaser
          </a>
        </li>
      </ul>
    </div>
    <div
      class="bb-show-page__season"
    >
      <h3
        class="bb-show-page__season-title"
      >
        <span>
          Season:
        </span>
        2
      </h3>
      <ul
        class="bb-show-page__season-list"
      >
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1076814"
          >
            The Last Donnycorn
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1089606"
          >
            Green Wing
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1090381"
          >
            15 Minutes of Fame
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1100163"
          >
            Splitsville
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1104815"
          >
            Clawdad
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1156710"
          >
            Super Sweet 6
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1156711"
          >
            A Star Is Blossom
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1156862"
          >
            Mini Golf Madness
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1156863"
          >
            Summer Bummer
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1156864"
          >
            The Tell Tale Schedulebot
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1201866"
          >
            Musclecup
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1201867"
          >
            Take Your Kids To Dooms Day
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1201868"
          >
            The Bubbles-sitters Club
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1214953"
          >
            Buttercup vs. Math
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1214954"
          >
            Home, Sweet Homesick
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1214955"
          >
            Memory Lane of Pain
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1214956"
          >
            Spider Sense
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1233374"
          >
            Imagine That
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1260256"
          >
            Phantasm Chasm
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1260257"
          >
            Tooth Or Consequences
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1277371"
          >
            Monkey Love
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1284764"
          >
            Bridezilla
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1289834"
          >
            Power of Four: Find Your Bliss
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1318440"
          >
            Power of Four: Bliss Reminisce
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1318441"
          >
            Power of Four: Blisster Sister
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1318442"
          >
            Power of Four: Breaking Bliss
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1318447"
          >
            Power of Four: Blisstersweet Symphony
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1319192"
          >
            Midnight At The Mayor's Mansion
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1355628"
          >
            You're A Good Man, Mojo Jojo
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1355629"
          >
            The Trouble With Bubbles
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1430717"
          >
            Never Been Blissed
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1430719"
          >
            Sugar, Spice, and Super Lice
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1430720"
          >
            The Buttercup Job
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1430721"
          >
            A Slight Hiccup
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1430722"
          >
            Toy Ploy
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1430723"
          >
            Derby Dollies
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1430724"
          >
            Bubbles the Blue
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1454433"
          >
            Deb O'Nair
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1471719"
          >
            Man Up 3: The Good, The Bad, and the Manly
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1454436"
          >
            The Blossom Files
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1471716"
          >
            Aliver
          </a>
        </li>
      </ul>
    </div>
    <div
      class="bb-show-page__season"
    >
      <h3
        class="bb-show-page__season-title"
      >
        <span>
          Season:
        </span>
        3
      </h3>
      <ul
        class="bb-show-page__season-list"
      >
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1454434"
          >
            Not So Secret Service
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1454435"
          >
            Worship
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1471715"
          >
            Blossom3
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1471717"
          >
            Mojo the Great
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1471709"
          >
            Trouble Clef
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1471710"
          >
            Save the Date
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1471711"
          >
            Can't Buy Love
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1471712"
          >
            Largo
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1471713"
          >
            Blundercup
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1471714"
          >
            Ragnarock and Roll
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1502843"
          >
            Total Eclipse Of The Kart, Parts 1 & 2
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1502844"
          >
            The Long Skate Home
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1502845"
          >
            Quarantine
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1502846"
          >
            In The Doghouse
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1502847"
          >
            Salamander
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1518396"
          >
            Small World: Abra-disaster
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1518397"
          >
            Small World: Stone Cold Spider
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1518398"
          >
            Small World: Maze Daze
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1518399"
          >
            Small World: Heart To Heartstone
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1536459"
          >
            Witch's Crew
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1567719"
          >
            The Gift
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1615024"
          >
            Oh, Daisy!
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1615025"
          >
            Brain Freeze
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1615026"
          >
            Lights Out!
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1615027"
          >
            Bucketboy!
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1615028"
          >
            The Fog
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1615029"
          >
            The Spoon
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1629436"
          >
            Cat Burglar
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1629437"
          >
            Hustlecup
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1629438"
          >
            Rebel Rebel
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1629439"
          >
            Our Brand is Chaos
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1629440"
          >
            Tagalong
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1629441"
          >
            Brainlord
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1629442"
          >
            Checkin' Out!
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1629443"
          >
            Drama Bomb
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1643954"
          >
            Watch It!
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1643951"
          >
            Man Up 4: The Donnyest Game
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1662009"
          >
            The Oct-father
          </a>
        </li>
        <li
          class="bb-show-page__episode"
        >
          <a
            class="bb-show-page__episode-link"
            href="/episodes/1662010"
          >
            Sideline Dad
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
